### PR TITLE
faml-660-matomo-psc-discrepancies-report

### PR DESCRIPTION
**Pull request by ICraig-CH**
-add matomo scripts to report part of the journey
-add onclick events for the text entry to get the value

Jira ticket: https://companieshouse.atlassian.net/browse/FAML-660

### CODE REVIEW

**Review by CH-JonathanBlake — APPROVED** [review]

File: server/views/report/contact_name.njk [hunk 1]
```diff
@@ -26,7 +26,7 @@
                     <input class="govuk-input govuk-!-width-two-thirds{{ ' govuk-input--error' if this_errors is not undefined }}" value="{{ this_data.fullName }}" id="full-name" name="fullName" type="text" autocomplete="name" spellcheck="false" />
                   </div>
 
-                  <button class="govuk-button" data-module="govuk-button">
+                  <button class="govuk-button" data-module="govuk-button" onclick="_paq.push(['trackEvent','5mld', 'What is your name?', 'Contact Name: ' + document.getElementById('full-name').value]);">
```
Comment (Zain-Abbas1) [inline server/views/report/contact_name.njk:29]: I thought there was something about sending data to Matomo, which is why we have the ability to mask sensitive data?

File: server/views/report/company_confirmation.njk [hunk 2]
```diff
@@ -48,7 +48,7 @@
                         </div>
                     </dl>
 
-                <button class="govuk-button" data-module="govuk-button">
+                <button class="govuk-button" data-module="govuk-button" onclick="_paq.push(['trackEvent','5mld', 'What is the name of your organisation?', 'Company with number: {{ this_data.company_number}} confirmed']);" >
```
Comment (Zain-Abbas1) [inline server/views/report/company_confirmation.njk:51]: I thought there was something about sending data to Matomo, which is why we have the ability to mask sensitive data?

Comment (Zain-Abbas1) [reply server/views/report/company_confirmation.njk]: Also, this page is confirmation of company details, the event suggests it's just the name that's relevant `What is the name of your organisation?`?

Comment (ICraig-CH) [reply server/views/report/company_confirmation.njk]: I just pushed up a change for that

File: server/views/report/contact_name.njk (hunk 1 above)
Comment (Zain-Abbas1) [reply server/views/report/contact_name.njk:29]: Speaking with Iain offline, the ticket specifically asks for the Matomo event to capture the contact name.

**Review by Zain-Abbas1 — APPROVED** [review]
👍

**Pull request merged**